src/qt3utils/datagenerators/princeton.py

```diff
@@ -0,0 +1,313 @@
+
+    @center_wavelength.setter
+    def center_wavelength(self, nanometers):
+        """
+        Sets the spectrometer center wavelength to nanometers.
+        """
+        #NOTE: The line below addresses bug where if step and glue is selected, doesn't allow setting center wavelength
+        self.light.set(lf.AddIns.ExperimentSettings.StepAndGlueEnabled, False)
+        return self.light.set(lf.AddIns.SpectrometerSettings.GratingCenterWavelength, nanometers)
+
+    @property
+    def experiment_name(self):
+        """
+        Returns the experiment name.
+        """
+        return self._experiment_name
+
+    @experiment_name.setter
+    def experiment_name(self, a_name):
+        """
+        User can set the experiment that they want to load.
+        """
+        if a_name != self._experiment_name:
+            self._experiment_name = a_name
+            self.light.load_experiment(self._experiment_name)
+
+    @property
+    def grating(self):
+        """
+        Returns the current grating.
+        """
+        return self.light.get(lf.AddIns.SpectrometerSettings.GratingSelected)
+
+    @grating.setter
+    def grating(self, grating_string):
+        """
+        Sets the current grating to be the one specified by parameter grating.
+        """
+        if grating_string in self.gratings_options:
+            self.light.set(lf.AddIns.SpectrometerSettings.GratingSelected, grating_string)
+        else:
+            logger.error(f"Grating {grating_string} is not an options. The options are: {self.gratings_options}")
+
+    @property
+    def gratings_options(self):
+        """
+        Returns a list of all installed gratings.
+        """
+        #NOTE: This line below is important. "GetCurrentCapabilities" is able to return the list of possibilities of any Lightfield call in order to provide more information.
+        available_gratings = self.light.experiment.GetCurrentCapabilities(lf.AddIns.SpectrometerSettings.GratingSelected)
+        return [str(a) for a in available_gratings]
+
+    @property
+    def num_frames(self):
+        """
+        Returns the number of frames taken during the acquisition.
+        """
+        return self.light.get(lf.AddIns.ExperimentSettings.FrameSettingsFramesToStore)
+
+    @num_frames.setter
+    def num_frames(self, num_frames):
+        """
+        Sets the number of frames to be taken during acquisition to number.
+        """
+        return self.light.set(lf.AddIns.ExperimentSettings.FrameSettingsFramesToStore, num_frames)
+
+    @property
+    def exposure_time(self):
+        """
+        Returns the single frame exposure time (in ms).
+        """
+        return self.light.get(lf.AddIns.CameraSettings.ShutterTimingExposureTime)
+
+    @exposure_time.setter
+    def exposure_time(self, ms):
+        """
+        Sets the single frame exposure time to be ms (in ms).
+        """
+        return self.light.set(lf.AddIns.CameraSettings.ShutterTimingExposureTime, ms)
```

Comment:
returns data?

Reply:
This is what set does, don't think I have to return anything. Mistake on my part. 

PS: Additionally the Stop method right below SetValue in the picture that is how I think we should implement the stop/pause functionality for the spectrometer

![image](https://github.com/qt3uw/qt3-utils/assets/19673888/af1f07b8-3021-43f4-a71f-b316dc9a200f)